packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -1232,6 +1241,134 @@ export default{
         [32m%s[0m If you think this is a bug then please create an issue at https://github.com/cloudflare/wrangler2/issues/new."
       `);
     });
+
+    it("should expire uploaded assets that aren't included anymore", async () => {
+      const assets = [
+        { filePath: "assets/file-1.txt", content: "Content of file-1" },
+        { filePath: "assets/file-2.txt", content: "Content of file-2" },
+      ];
+      const kvNamespace = {
+        title: "__test-name-workers_sites_assets",
+        id: "__test-name-workers_sites_assets-id",
+      };
+      writeWranglerToml({
+        main: "./index.js",
+        site: {
+          bucket: "assets",
+        },
+      });
+      writeWorkerSource();
+      writeAssets(assets);
+      mockUploadWorkerRequest();
+      mockSubDomainRequest();
+      mockListKVNamespacesRequest(kvNamespace);
+      mockKeyListRequest(
+        kvNamespace.id,
+        [
+          // Put file-1 in the KV namespace
+          "assets/file-1.2ca234f380.txt",
+          // As well as a couple from a previous upload
+          "assets/file-3.somehash.txt",
+          "assets/file-4.anotherhash.txt",
+        ],
+        100,
+        undefined,
+        // disable expiration values for this test
+        null
+      );
+
+      // Check that we pull down the values of file-3.txt and file-4.txt
+      setMockFetchKVGetValue(
+        "some-account-id",
+        "__test-name-workers_sites_assets-id",
+        "assets/file-3.somehash.txt",
+        btoa("Content of file-3")
+      );
+      setMockFetchKVGetValue(
+        "some-account-id",
+        "__test-name-workers_sites_assets-id",
+        "assets/file-4.anotherhash.txt",
+        btoa("Content of file-4")
+      );
+
+      // and send them back up
+      mockUploadAssetsToKVRequest(kvNamespace.id, [
+        ...assets.filter((a) => a.filePath !== "assets/file-1.txt"),
+        {
+          filePath: "assets/file-3.txt",
+          content: "Content of file-3",
+          expiration: Date.now() + 5 * 60 * 3000,
```

Comment:
The `mockUploadAssetsToKVRequest()` helper is not actually checking that these values are correct. And I think that this test would fail if it did.
I think the value here should be `(Date.now() + 5*60*1000)/1000`.
But also I think we need to mock out `Date.now()` to avoid flaky tests due to the possible time difference between setting these values here and computing the values in the real code.

Reply:
noted noted

Reply:
done